front-end/src/domains/game-board/components/game-grid/game-grid.component.tsx

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { hexSize, placeholderBoard, size } from "./game-grid.constant";
+import { Grid } from "./game-grid.types";
+import { updateGridWithRandomLandTypes } from "./game-grid.utils";
+
+import { GameHex } from "./game-hex.component";
+
+/**
+ * The Board component renders the game board.
+ * It takes a savedBoard prop, which is a JSON string representing the initial state of the board.
+ * If no saved board is provided, it falls back to a placeholder board.
+ */
+interface BoardProps {
+  savedBoard?: string; // Optional prop for the initial state of the board
+}
+
```

Comment:
I didn't move this Props interface outside of the component - prop interfaces are normally in the component files, so it's appropriate to leave this here 🔥